challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -12434,6 +12434,1022 @@
+        },
+        {
+          "text":
+            "'<code>pascal(11,4,40,151)</code> should return <code>'+JSON.stringify(results[0])+'</code>.'",
+          "testString":
+            "assert.deepEqual(pascal(11,4,40,151),results[0],'<code>pascal(11,4,40,151)</code> should return <code>'+JSON.stringify(results[0])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>pascal(11,4,41,151)</code> should return <code>'+JSON.stringify(results[1])+'</code>.'",
+          "testString":
+            "assert.deepEqual(pascal(11,4,41,151),results[1],'<code>pascal(11,4,41,151)</code> should return <code>'+JSON.stringify(results[1])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>pascal(11,4,35,151)</code> should return <code>'+JSON.stringify(results[2])+'</code>.'",
+          "testString":
+            "assert.deepEqual(pascal(11,4,35,151),results[2],'<code>pascal(11,4,35,151)</code> should return <code>'+JSON.stringify(results[2])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>pascal(11,4,440,151)</code> should return <code>'+JSON.stringify(results[3])+'</code>.'",
+          "testString":
+            "assert.deepEqual(pascal(11,4,440,151),results[3],'<code>pascal(11,4,440,151)</code> should return <code>'+JSON.stringify(results[3])+'</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f5e",
+      "challengeType": 5,
+      "releasedOn": "August 15, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": [
+            "function pascal (a, b, mid, top) {",
+            "  // Good luck!",
+            "}"
+          ],
+          "head": [],
+          "tail": [
+            "var results=[",
+            "[5,13,8],",
+            "[6,13,7],",
+            "[0,13,13],",
+            "[405,13,-392]]"
+          ]
+        }
+      }
+    },
+    {
+      "title": "Pascal's triangle",
+      "description": [
+        "<a href=\"https://en.wikipedia.org/wiki/Pascal's triangle\">Pascal's triangle</a>  is an arithmetic and geometric figure first imagined by  <a href=\"https://en.wikipedia.org/wiki/Blaise Pascal\">Blaise Pascal</a>.",
+        "Its first few rows look like this:",
+        "1",
+        "1 1",
+        "1 2 1",
+        "1 3 3 1",
+        "where each element of each row is either 1 or the sum of the two elements right above it.",
+        "For example, the next row of the triangle would be:",
+        "<b>1</b>  (since the first element of each row doesn't have two elements above it)</span>",
+        "<b>4</b>  (1 + 3)</span>",
+        "<b>6</b>  (3 + 3)</span>",
+        "<b>4</b>  (3 + 1)</span>",
+        "<b>1</b>  (since the last element of each row doesn't have two elements above it) </span>",
+        "So the triangle now looks like this:",
+        "1",
```

Comment:
And for this triangle:

`"<pre>    1\n   1 1\n  1 2 1\n 1 3 3 1\n1 4 6 4 1</pre>"`